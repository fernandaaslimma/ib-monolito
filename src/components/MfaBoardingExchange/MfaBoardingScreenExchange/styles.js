import styled, { css } from "styled-components";
import { rem, media } from "../../../styles/tools";

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: auto;

  ${media.md(css`
    padding-left: ${props => rem(props.paddingLeft)};
    padding-right: ${props => rem(props.paddingRight)};
  `)};
`;
