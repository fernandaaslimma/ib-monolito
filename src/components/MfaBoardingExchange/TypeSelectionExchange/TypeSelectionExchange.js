import React from "react";
import ClickWrapper from "../../../utils/clickWrapper";
import Icon from "../../common/Icon";
import {
  Wrapper,
  WrapperTitle,
  Title,
  SubTitle,
  WrapperSelectionType,
  SelectionType,
  SelectionTypeText
} from "./styles.js";
import { translate } from "../../../utils/i18n";
import { MFABOARDING_PARAM_AUTH } from "../../../utils/constants";

function TypeSelectionExchange({ changeCurrentScreen, notification }) {
  return (
    <Wrapper>
      <WrapperTitle data-test="MfaExchangeSelectionTypeTitle">
        <Title fontSize={22} marginBottom={12}>
          {translate("TYPE_SELECTION_TITLE_EX")}
        </Title>
        <SubTitle fontSize={18}>
          {translate("TYPE_SELECTION_SUBTITLE_EX")}
        </SubTitle>
      </WrapperTitle>

      <WrapperSelectionType>
        {notification.parameters.map((item, index) => (
          <ClickWrapper key={index}>
            <SelectionType
              hasMarginBottom={
                notification.parameters.lenght === Number(index) + 1
                  ? false
                  : true
              }
              onClick={() => {
                changeCurrentScreen(2);
              }}
              data-test={`MfaExchangeSelectionTypeBtn_${item.id}`}
            >
              {item.type === MFABOARDING_PARAM_AUTH && (
                <Icon type="MfaAuthenticator" />
              )}

              <SelectionTypeText>
                <Title fontSize={16} marginBottom={8}>
                  {translate(`TYPE_SELECTION_BUTTON_${item.id}_TITLE`)}
                </Title>
                <SubTitle fontSize={14}>
                  {translate("TYPE_SELECTION_BUTTON_SUBTITLE_EX")}
                </SubTitle>
              </SelectionTypeText>
              <Icon type="MfaChevronRight" />
            </SelectionType>
          </ClickWrapper>
        ))}
      </WrapperSelectionType>
    </Wrapper>
  );
}
export default TypeSelectionExchange;
