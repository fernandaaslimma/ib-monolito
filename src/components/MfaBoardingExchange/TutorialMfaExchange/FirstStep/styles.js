import styled from "styled-components";
import { remFontSize, rem } from "../../../../styles/tools";
import { blue70, lightBlack } from "../../../../styles/settings";

export const Title = styled.div`
  font-family: Lato;
  font-size: ${remFontSize(22)};
  font-weight: bold;
  letter-spacing: 0.78px;
  text-align: center;
  color: ${blue70};
  margin-top: ${rem(33)};
  margin-bottom: ${rem(33)};
`;

export const MfaImgTutorialText = styled.span`
  width: 30%;
  font-family: Lato;
  font-size: ${remFontSize(15)};
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  letter-spacing: 0.35px;
  color: ${lightBlack};
  text-align: left;
`;

export const MfaTextNumber = styled.span`
  font-weight: 900;
  color: ${blue70};
`;

export const Container = styled.div`
  width: 100%;
  max-width: ${rem(500)};
  margin: 0 auto;

  @media (min-width: 500px) {
    min-width: ${rem(500)};
  }
`;

export const Image = styled.img`
  max-width: ${rem(250)};
  max-height: ${rem(250)};
  width: 70%;
  display: block;
  margin: 0 auto;
  margin-bottom: ${rem(33)};
`;

export const TextDiv = styled.div`
  width: 70%;
  display: block;
  margin: 0 auto;
`;
