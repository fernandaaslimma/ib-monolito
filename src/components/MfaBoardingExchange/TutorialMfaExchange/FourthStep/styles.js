import styled from "styled-components";
import { remFontSize, rem } from "../../../../styles/tools";
import { blue70, lightBlack } from "../../../../styles/settings";

export const SpanWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  margin-bottom: ${rem(66)};
  margin-top: ${rem(66)};
  padding-inline: ${rem(16)};
`;

export const Span = styled.div`
  font-weight: ${rem(300)};
  font-family: Lato;
  color: #587485;
  letter-spacing: ${rem(0.45)};
  font-size: ${remFontSize(18)};
  inline-size: ${rem(312)};
  overflow-wrap: break-word;
`;

export const Title = styled.div`
  font-family: Lato;
  font-size: ${remFontSize(22)};
  font-weight: bold;
  letter-spacing: 0.78px;
  text-align: center;
  color: ${blue70};
  margin-top: ${rem(33)};
  margin-bottom: ${rem(33)};
`;

export const MfaImgTutorialText = styled.span`
  width: 30%;
  font-family: Lato;
  font-size: ${remFontSize(15)};
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  letter-spacing: 0.35px;
  color: ${lightBlack};
  text-align: left;
`;

export const MfaTextNumber = styled(MfaImgTutorialText)`
  font-weight: 900;
  color: ${blue70};
`;

export const Container = styled.div`
  width: 100%;
  max-width: ${rem(500)};
  margin: 0 auto;

  @media (min-width: 500px) {
    min-width: ${rem(500)};
  }
`;

export const ContainerTwoButtons = styled.div`
  width: 90%;
  max-width: ${rem(500)};
  margin: 0 auto;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const Image = styled.img`
  max-width: ${rem(250)};
  max-height: ${rem(250)};
  width: 70%;
  display: block;
  margin: 0 auto;
  margin-bottom: ${rem(33)};
`;

export const TextDiv = styled.div`
  width: 70%;
  display: block;
  margin: 0 auto;
`;
