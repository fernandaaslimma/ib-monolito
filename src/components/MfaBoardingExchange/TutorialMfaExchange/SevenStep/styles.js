import styled from "styled-components";
import { remFontSize, rem } from "../../../../styles/tools";
import { blue70, lightBlack, black50 } from "../../../../styles/settings";

export const SubTitle = styled.p`
  font-family: Lato;
  font-size: ${remFontSize(22)};
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.32;
  letter-spacing: ${rem(0.58)};
  color: ${black50};
  margin-bottom: ${rem(20)};
`;

export const Title = styled.div`
  font-family: Lato;
  font-size: ${remFontSize(22)};
  font-weight: bold;
  letter-spacing: 0.78px;
  text-align: center;
  color: ${blue70};
  margin-top: ${rem(33)};
  margin-bottom: ${rem(33)};
`;

export const MfaImgTutorialText = styled.span`
  width: 30%;
  font-family: Lato;
  font-size: ${remFontSize(15)};
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  letter-spacing: 0.35px;
  color: ${lightBlack};
  text-align: left;
`;

export const MfaTextNumber = styled(MfaImgTutorialText)`
  font-weight: 900;
  color: ${blue70};
`;

export const Container = styled.div`
  width: 100%;
  max-width: ${rem(500)};
  margin: 0 auto;

  @media (min-width: 500px) {
    min-width: ${rem(500)};
  }
`;

export const Image = styled.img`
  max-width: ${rem(250)};
  max-height: ${rem(250)};
  width: 20%;
  display: block;
  margin: 0 auto;
  margin-bottom: ${rem(33)};
  margin-top: ${rem(66)};
`;

export const TextDiv = styled.div`
  width: 70%;
  display: block;
  margin: 0 auto;
`;
