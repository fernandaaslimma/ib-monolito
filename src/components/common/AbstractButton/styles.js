import styled, { css } from "styled-components";
import { rem, media } from "../../../styles/tools";
import {
  lightBlue,
  darkBlue,
  green,
  white,
  grey,
  lightRed,
  lighterRed,
  grey70
} from "../../../styles/settings";

export const ButtonTag = styled.button`
  display: flex;
  justify-content: center;
  font-family: 'Lato Bold', 'Lato';
  background: none;
  /* border-radius: ${rem(4)}; */
  border: 0;
  border-radius: 4px;
  cursor: pointer;
  font-size: ${rem(12)};
  letter-spacing: ${rem(0.2)};
  line-height: ${rem(45)};
  padding: ${props => (props.noPadding ? "0 rem(20)" : 0)};
  position: relative;
  text-align: center;
  text-decoration: none;
  /* box-shadow: 0 4px 7px 0 rgba(74, 144, 226, 0.3); */
  transition: 0.2s ease-in-out color, 0.2s ease-in-out background,
    0.2s ease-in-out box-shadow, 0.2s ease-in-out border, 0.1s transform;
  white-space: nowrap;
  width: ${props => (props.width ? rem(props.width) : "100%")};
  user-select: none;
  /* opacity: ${props =>
    props.opacity && props.disabled ? props.opacity : 1}; */

  ${({ width, isPending }) =>
    width &&
    isPending &&
    css`
      min-width: ${rem(width)} !important;
      margin-right: 0 !important;
      text-align: center;
      padding: 0 ${rem(10)} !important;

      ${media.lg(css`
        margin-right: ${rem(20)} !important;
      `)};
    `};

  ${media.md(css`
    width: ${props => (props.width ? rem(props.width) : "100%")};
  `)};

  ${media.lg(css`
    font-size: ${rem(13)};
    padding: 0 ${rem(28)};
    line-height: ${rem(42)};
  `)};

  ${({ noPadding }) =>
    noPadding &&
    media.lg(css`
      padding: 0;
    `)}

  :active {
    transform: translateY(${rem(1)});
  }

  ${({ isCallToAction }) =>
    isCallToAction &&
    css`
      background-color: ${green};
      border: solid ${rem(1)} ${green};
      box-shadow: 0 ${rem(2)} ${rem(7)} ${rem(2)} rgba(117, 214, 184, 0.39);
      color: ${white};
      font-family: "Lato Bold", Lato;

      :hover:not([disabled]) {
        background-color: ${lightBlue};
        border-color: ${lightBlue};
        box-shadow: 0 ${rem(2)} ${rem(7)} ${rem(2)} rgba(51, 204, 208, 0.39);
      }
    `};

  ${({ isWarning }) =>
    isWarning &&
    css`
      background-color: ${lightRed};
      border: solid ${rem(1)} ${lightRed};
      box-shadow: 0 ${rem(2)} ${rem(7)} ${rem(2)} rgba(233, 103, 103, 0.3);
      color: ${white};

      :hover:not([disabled]) {
        background-color: ${lighterRed};
        border-color: ${lighterRed};
      }
    `};

  ${({ isCallToAction, isWarning }) =>
    !isCallToAction &&
    !isWarning &&
    css`
      /* border: solid ${rem(1)} ${darkBlue};
      color: ${white}; */

      :hover:not([disabled]) {
        opacity: 0.7;
      }
    `};

  ${({ small }) =>
    small &&
    css`
      font-size: ${rem(11)} !important;
      height: ${rem(40)} !important;
      line-height: ${rem(40)} !important;
    `};

  ${({ disabled, loading, opacity }) =>
    disabled &&
    !loading &&
    opacity &&
    css`
      background-color: ${!opacity && grey};
      opacity: ${opacity};
      box-shadow: none;
      border: solid ${rem(1)} ${grey};
      color: ${white};
      cursor: not-allowed;
      pointer-events: none;
    `};

  ${({ disabled, loading }) =>
    disabled &&
    loading &&
    css`
      box-shadow: none;
      cursor: not-allowed;
      pointer-events: none;
    `};
  ${({ disabled, loading }) =>
    disabled &&
    !loading &&
    css`
      background-color: ${grey};
      box-shadow: none;
      border: solid ${rem(1)} ${grey};
      color: ${white};
      cursor: not-allowed;
      pointer-events: none;
    `};

  ${({ disabled, loading }) =>
    disabled &&
    loading &&
    css`
      box-shadow: none;
      cursor: not-allowed;
      pointer-events: none;
    `};

  ${({ isSecondary }) =>
    isSecondary &&
    css`
      background-color: ${white};
      border-color: ${darkBlue};
      color: ${darkBlue};
      :hover:not([disabled]) {
        background-color: ${darkBlue};
        border-color: ${darkBlue};
        color: ${white};
      }
    `}

  ${({ isSpanButton }) =>
    isSpanButton &&
    css`
      background-color: transparent;
      border: unset;
      color: ${grey70};
      box-shadow: unset;

      &:hover {
        background-color: unset !important;
        color: unset !important;
        cursor: text;
      }

      &:focus {
        outline: unset !important;
      }

      &:active {
        transform: unset !important;
      }
    `};

  ${({ heightPx }) =>
    heightPx &&
    css`
      line-height: ${rem(heightPx)};
      height: ${rem(heightPx)};
    `}
`;
