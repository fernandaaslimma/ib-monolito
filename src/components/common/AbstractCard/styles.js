import styled, { css } from "styled-components";

import { white } from "../../../styles/settings";
import { rem } from "../../../styles/tools";

export const CardWrapper = styled.div`
  border-radius: ${rem(4)};
  background: ${white};
  padding: ${rem(16)} ${rem(30)};
  margin-top: ${({ marginTop }) => marginTop};
  margin-bottom: ${({ marginBottom }) => marginBottom};
  box-shadow: 0px 2px 22px 2px rgba(225, 237, 246, 0.5);
  max-width: 100%;
  position: relative;

  ${({ height }) =>
    height &&
    css`
      height: ${height};
    `};
`;
