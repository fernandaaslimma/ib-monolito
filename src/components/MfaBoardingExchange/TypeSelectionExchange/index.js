import TypeSelectionExchange from "./TypeSelectionExchange";

export default TypeSelectionExchange;
