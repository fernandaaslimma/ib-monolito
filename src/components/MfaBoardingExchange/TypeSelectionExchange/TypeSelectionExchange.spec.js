import React from "react";
import { shallow } from "enzyme";
import ClickWrapper from "../../../utils/clickWrapper";
import { SelectionType } from "./styles";
import {
  MFABOARDING_PARAM_APP,
  MFABOARDING_PARAM_AUTH
} from "../../../utils/constants";
import TypeSelectionExchange from "./TypeSelectionExchange";

const props = {
  current: 1,
  changeCurrentScreen: jest.fn(),
  notification: {
    title: "Precisamos cadastrar um novo fator de autenticação",
    description: "Qual fator deseja cadastrar?",
    displayMethod: "PopUp",
    type: "MFABoarding",
    parameters: [
      {
        type: MFABOARDING_PARAM_APP,
        id: 1
      }
    ]
  }
};

describe("TypeSelection component", () => {
  it("should match snapshot with props", () => {
    expect(shallow(<TypeSelectionExchange {...props} />)).toMatchSnapshot();
  });

  it("should call changeCurrentScreen with type authenticatorApp", () => {
    const newProps = {
      ...props,
      notification: {
        title: "Precisamos cadastrar um novo fator de autenticação",
        description: "Qual fator deseja cadastrar?",
        displayMethod: "PopUp",
        type: "MFABoarding",
        parameters: [
          {
            type: MFABOARDING_PARAM_AUTH,
            id: 2
          }
        ]
      }
    };
    const shallowComponent = shallow(<TypeSelectionExchange {...newProps} />);

    shallowComponent
      .find(ClickWrapper)
      .find(SelectionType)
      .prop("onClick")();
    expect(props.changeCurrentScreen).toHaveBeenCalledWith(2);
  });
});
