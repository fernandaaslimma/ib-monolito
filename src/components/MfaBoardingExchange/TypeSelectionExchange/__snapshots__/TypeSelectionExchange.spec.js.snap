// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeSelection component should match snapshot with props 1`] = `
<styled.div>
  <styled.div
    data-test="MfaExchangeSelectionTypeTitle"
  >
    <styled.p
      fontSize={22}
      marginBottom={12}
    />
    <styled.p
      fontSize={18}
    />
  </styled.div>
  <styled.div>
    <ConnectWrapper
      key="0"
    >
      <styled.div
        data-test="MfaExchangeSelectionTypeBtn_1"
        hasMarginBottom={true}
        onClick={[Function]}
      >
        <styled.div>
          <styled.p
            fontSize={16}
            marginBottom={8}
          />
          <styled.p
            fontSize={14}
          />
        </styled.div>
        <Icon
          color={null}
          cursor={false}
          fixCrossAxisAlign={null}
          height={null}
          type="MfaChevronRight"
          width={null}
        />
      </styled.div>
    </ConnectWrapper>
  </styled.div>
</styled.div>
`;
